src/datafusion/table_provider.rs

```diff
@@ -99,25 +101,46 @@ impl TableProvider for ZarrTableProvider {
         &self,
         filters: &[&Expr],
     ) -> datafusion::error::Result<Vec<TableProviderFilterPushDown>> {
-        // TODO: which filters can we push down?
         Ok(filters
             .iter()
-            .map(|_| TableProviderFilterPushDown::Unsupported)
+            .map(|filter| {
+                if expr_applicable_for_cols(
+                    &self
+                        .table_schema
+                        .fields
+                        .iter()
+                        .map(|field| field.name().to_string())
+                        .collect::<Vec<_>>(),
+                    filter,
+                ) {
+                    TableProviderFilterPushDown::Exact
```

Comment:
I know you made reference to this previously, but could you say again why you're implementing `Exact` + implementing the record batch evaluation and not `Inexact` and letting DataFusion handle winnowing down the larger chunks from zarr? To your other point, that pushdown to the initial chunk seems to be the value add.

Reply:
That's right, I should have made the comment here instead of a general comment. It's currently like that because tbh I didn't completely understand how this works when I first wrote it. I plan on changing this to `Inexact` (and I'll only use `Exact` for hive style partitioning), but that will require a few changes in the reader as well, to drop some of the more convoluted row filtering logic I wrote, and since I've been working on this for a while I wanted to put up a work in progress PR to start gathering your feedback while I make that change. 

(well, technically I could change this to `Inexact` right now and it would still work of course, but I wanted to cleanly update the reader and the pushdown type at the same time).

Reply:
Fixed now.